spec/support/git_spec_helpers.rb

```diff
@@ -23,6 +23,41 @@ def repo(options = {})
     end
   end
 
+  # Creates a local repository in 'local-repo' folder with a remote repository
+  # which points to 'some-repo' folder. The current working directory is set to
+  # the local repository's root directory.
+  def with_remote_repo
+    repo do
+      touch 'update-me'
+      `echo update >> update-me`
+      touch 'delete-me'
+      `echo delete >> delete-me`
+      `git add . 2>&1 > #{File::NULL}`
+      `git commit -m "Initial commit" 2>&1 > #{File::NULL}`
+
+      Dir.chdir '..' do
+        remote_url = "file://#{File.expand_path('some-repo')}"
+        `git clone "#{remote_url}" local-repo 2>&1 > #{File::NULL}`
+      end
+
+      Dir.chdir '../local-repo' do
+        yield if block_given?
```

Comment:
Not to be overly pedantic, but when I read `with_remote_repo` I would expect that the block I pass would execute within the context of the remote repository, but I'll admit it's a bit hard to describe what we're trying to do concisely.